controllers/administration.js

```diff
@@ -2345,43 +2345,26 @@ const getCourseCreateHandler = () => function coruseCreateHandler(req, res, next
 		});
 };
 
-const updateSchoolFeature = async (req, currentFeatures, newState, featureName) => {
-	const isCurrentlyAllowed = (currentFeatures || []).includes(featureName);
-
-	if (!isCurrentlyAllowed && newState) {
-		// add feature
-		await api(req)
-			.patch(`/schools/${req.params.id}`, {
-				json: {
-					$push: {
-						features: featureName,
-					},
-				},
-			});
+const updateSchoolFeatures = async (req, currentFeatures, features) => {
+	const pushValues = Object.keys(features)
+		.filter((feature) => features[feature] && !currentFeatures.includes(feature));
+	const pullValues = Object.keys(features)
+		.filter((feature) => !features[feature] && currentFeatures.includes(feature));
+
+	const requests = [];
+	if (pushValues.length > 0) {
+		requests.push(api(req)
+			.patch(`/schools/${req.params.id}`, { json: { $push: { features: { $each: pushValues } } } }));
 	}
-
-	if (isCurrentlyAllowed && !newState) {
-		// remove feature
-		await api(req)
-			.patch(`/schools/${req.params.id}`, {
-				json: {
-					$pull: {
-						features: featureName,
-					},
-				},
-			});
+	if (pullValues.length > 0) {
+		requests.push(api(req)
+			.patch(`/schools/${req.params.id}`, { json: { $pull: { features: { $in: pullValues } } } }));
```

Comment:
We add a feature route for it or not?

Reply:
The feature route is only implemented for messenger settings: https://github.com/hpi-schul-cloud/schulcloud-server/blob/develop/src/services/messenger/services/messengerConfigService.js

In the administration request in the client other flags are set as well, so we handle all of them together.